### PR TITLE
feat: support for calendar event rsvps

### PR DESCRIPTION
**Pull request by AusKarlos87**
Please describe the changes this PR makes and why it should be merged:
Added all rsvp websocket events and methods associated with creating/updating and deleting rsvps via the bot API.
Rsvps are cached (if enabled) in a collection with their associated cached calendar event

Status

-   [x] Code changes have been tested.

Semantic versioning classification:

-   [x] This PR changes the library's interface (methods or parameters added)
-   [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
-   [ ] This PR only includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/guilded.js/lib/gateway/ClientGatewayHandler.ts [hunk 1]
```diff
@@ -32,17 +32,21 @@ import type {
...
 import { ListEventHandler } from "./handler/ListEventHandler";
 import { TeamChannelEventHandler } from "./handler/TeamChannelEventHandler";
 import { DocEventHandler } from "./handler/DocEventHandler";
 import { ReactionEventHandler } from "./handler/ReactionEventHandler";
-import { CalendarEventHandler } from "./handler/CalendarEventHandler";
+import { CalendarEventHandler, CalendarEventRsvpHandler } from "./handler/CalendarEventHandler";
 
 export class ClientGatewayHandler {
     calendarEventHandler = new CalendarEventHandler(this.client);
+    CalendarEventRsvpHandler = new CalendarEventRsvpHandler(this.client);
```
Comment (zaida04) [inline packages/guilded.js/lib/gateway/ClientGatewayHandler.ts]:
```suggestion
    calendarEventRsvpHandler = new CalendarEventRsvpHandler(this.client);
```

File: packages/guilded.js/lib/gateway/ClientGatewayHandler.ts [hunk 2]
```diff
@@ -56,6 +60,9 @@ export class ClientGatewayHandler {
         [WebSocketEvents.CalendarEventCreated]: (data) => this.calendarEventHandler.calendarEventCreated(data as WSCalendarEventCreated),
         [WebSocketEvents.CalendarEventDeleted]: (data) => this.calendarEventHandler.calendarEventDeleted(data as WSCalendarEventDeleted),
         [WebSocketEvents.CalendarEventUpdated]: (data) => this.calendarEventHandler.calendarEventUpdated(data as WSCalendarEventUpdated),
+        [WebSocketEvents.CalendarEventRsvpUpdated]: (data) => this.CalendarEventRsvpHandler.calendarEventRsvpUpdated(data as WSCalendarEventRsvpUpdated),
+        [WebSocketEvents.CalendarEventRsvpManyUpdated]: (data) => this.CalendarEventRsvpHandler.calendarEventRsvpManyUpdated(data as WSCalendarEventRsvpManyUpdated),
+        [WebSocketEvents.CalendarEventRsvpDeleted]: (data) => this.CalendarEventRsvpHandler.calendarEventRsvpDeleted(data as WSCalendarEventRsvpDeleted),
```
Comment (zaida04) [inline packages/guilded.js/lib/gateway/ClientGatewayHandler.ts]:
```suggestion
        [WebSocketEvents.CalendarEventRsvpUpdated]: (data) => this.calendarEventRsvpHandler.calendarEventRsvpUpdated(data as WSCalendarEventRsvpUpdated),
        [WebSocketEvents.CalendarEventRsvpManyUpdated]: (data) => this.calendarEventRsvpHandler.calendarEventRsvpManyUpdated(data as WSCalendarEventRsvpManyUpdated),
        [WebSocketEvents.CalendarEventRsvpDeleted]: (data) => this.calendarEventRsvpHandler.calendarEventRsvpDeleted(data as WSCalendarEventRsvpDeleted),
```

File: packages/rest/lib/util/Router.ts [hunk 3]
```diff
@@ -149,6 +153,26 @@ export class Router {
...
+    }
+
+    /** Update a rsvp user from a calendar event */
+    updateCalendarEventRvsp(
+        channelId: string,
+        calendarEventId: number,
+        userId: string,
+        options: RESTPatchCalendarEventRsvpBody,
+    ): Promise<RESTPatchCalendarEventRsvpResult> {
+        return this.rest.put<RESTPatchCalendarEventRsvpResult>(ROUTES.calendarEventRsvp(channelId, calendarEventId, userId), options);
```
Comment (zaida04) [inline packages/rest/lib/util/Router.ts]:
```suggestion
        return this.rest.put<RESTPatchCalendarEventRsvpResult, RESTPatchCalendarEventRsvpBody>(ROUTES.calendarEventRsvp(channelId, calendarEventId, userId), options);
```

**Review by zaida04 — CHANGES_REQUESTED** [review]

**Review by zaida04 — APPROVED** [review]

**Comment by zaida04** [comment]
once you review, feel free to merge @DaStormer

**Review by DaStormer — APPROVED** [review]

**Pull request merged**